lib/mail/field.rb

```diff
@@ -145,7 +109,12 @@ def split(raw_field) #:nodoc:
       end
 
       def field_class_for(name) #:nodoc:
-        FIELDS_MAP[name.to_s.downcase]
+        name = name.to_s.downcase
+        @fields_map ||= {}
+        @fields_map[name] ||= begin
+          klass = name.split('-').map(&:capitalize).join + "Field"
+          Mail.const_get(klass) rescue nil
+        end
```

Comment:
Thanks @c960657 so much for this PR, I'd prefer to keep FEILDS_MAP though as , if we don't want to instantiate the class we could map the class names as a string, then in this method just call `Mail.const_get(FIELDS_MAP[name.to_s.downcase])`

Reason is for simplicity for future debugging and for class names that might not match the field header name for some reason in the future.

Reply:
If you can modify the above and ideally not delete the FIELD_MAP_NAME hash as well, I'll merge this for the next release.

Reply:
I just realised my latest commit wasn't quite right. I'll push an update later today.

Reply:
Done. I hope this is what you intended.

The new code is a bit more explicit, but I have added tests to help keep the large arrays/hashes in sync.

Reply:
Nice, thank you @c960657 :)